### ISSUE TITLE
The file is not displayed in the text editor because it is a directory.

### ISSUE THREAD

**Issue by menturion** [DEVSENSE/phptools-docs#707]
PHP v.8.0.

I just updated to the latest version `v1.52.16226` from a way older version. 

I am getting a large number of issues not being able to open the affected file when I double click on the message, showing the message “`The file is not displayed in the text editor because it is a directory`”.

**Comment by jakubmisek** [comment]
Thank you for trying the latest release!

Do you see the actual path of those files?
Is there anything similar to those issues/warnings?

**Comment by menturion** [comment]
The last version that works properly for me is `v1.47.15538 (pre-release)`.

When switching from that version to `v1.47.15561 (pre-release)` I get a huge number of issues (712 compared to 5 using v1.47.15538 (pre-release)) including the one(s) reported here, see the attached screenshot. This is the reason why I stopped updating the extension. 

![2024-10-25_163947](https://github.com/user-attachments/assets/f49a2808-9cb4-45b0-b4ab-8ebcd7cf26b2)

Nothing happens when I click on the buttons `"Open/Reveal Folder`".

**Comment by jakubmisek** [comment]
The path is definitely wrong (`C:\xampp`), but the warnings may be correct if you're running PHP >= 8.0.

May I ask, where is the class "HttpBasic" actually located? Isn't it some remote folder or anything unusual?

**Comment by menturion** [comment]
It is located within a subfolder inside the root `C:\xampp\htdocs` referenced via a class map similar to the composer autoload class map.

**Comment by jakubmisek** [comment]
so it is in `C:\xampp\htdocs\...` and in VSCode, you open the folder `C:\xampp\htdocs\` ?

**Comment by menturion** [comment]
> so it is in `C:\xampp\htdocs\...` and in VSCode, you open the folder `C:\xampp\htdocs\` ?

That's right.

(There must have been a change from `v1.52.16226 (pre-release)` to `v1.47.15561 (pre-release)` that caused this issue.)

**Comment by jakubmisek** [comment]
Is there anything special in the path? A non-ascii character maybe?

**Comment by menturion** [comment]
> Is there anything special in the path? A non-ascii character maybe?

No, nothing special. No special characters etc.
This is not the only affected class. The number of affected classes is about 100. 

The issues arose "ceteris paribus" from v1.47.15538 (pre-release) to v1.47.15561 (pre-release).

**Comment by jakubmisek** [comment]
There must be something in the path :/ otherwise, I don't know.

May I ask what is the path to one of the affected classes?

**Comment by menturion** [comment]
The entries in this class map looks like, e.g. ...

`'phpseclib3\\Crypt\\AES' => 'phpseclib/phpseclib/phpseclib/Crypt/AES.php',`

... whereas the absolute (full) path is built inside a "spl_autoload_register".

**Comment by jakubmisek** [comment]
So the path is: `c:/xampp/htdocs/lib/phpseclib/phpseclib/phpseclib/Crypt/AES.php`

and the VSCode Workspace is: `c:\xampp\htdocs\`

and VSCode is updated.

**Comment by menturion** [comment]
That's right.

To be completely correct, the path would be '`c:\xampp\htdocs\lib\phpseclib\`phpseclib/phpseclib/Crypt/AES.php', as the path to the root directory of the library is created taking the DIRECTORY_SEPARATOR into account. Replacing the slashes “/” with the backslash “\” (or vice versa) has no effect on the reported issue. 

If you already have an idea how to fix this issue, you can provide me with a hotfix so that I can make the changes locally and test them before you release a new version.

**Comment by jakubmisek** [comment]
I'm struggling with this one.

It seems VSCode breaks the path and shows `"c:\xampp"` instead, which is a totally bad thing.

I'll prepare a debug pre-release version, which adds some logging so we'll see what's happening.

**Comment by menturion** [comment]
Ok.

I would just like to add that if I copy the issue and not just the issue message, the following data is given as an example with respect to the  `"resource": "/c:/xampp/htdocs/"`:

```
[{
	"resource": "/c:/xampp/htdocs/",
	"owner": "_generated_diagnostic_collection_name_#2",
	"code": "PHP2439",
	"severity": 8,
	"message": "Declaration of HttpBasic::validateRequest(RequestInterface $request, ResponseInterface $response) must be compatible with ClientAssertionTypeInterface::validateRequest(RequestInterface $request, ResponseInterface $response): bool|null",
	"source": "PHP",
	"startLineNumber": 51,
	"startColumn": 5,
	"endLineNumber": 51,
	"endColumn": 92
}]
```

**Comment by jakubmisek** [comment]
Thank you. The `"resource"` property is invalid. We need to find where it comes from, and why it's trimmed.

**Comment by jakubmisek** [comment]
I see there are 712 problems in the incorrect "c:\xampp\" path.

<img width="252" alt="image" src="https://github.com/user-attachments/assets/dd50d006-da70-484f-8f65-86d424a6308f">

The remaining **5** problems, which have the correct path, also come from PHP? The affected files are currently opened?

**Comment by menturion** [comment]
Correct, the remaining 5 problems originate from PHP and have correct paths so that the corresponding files can be opened.

**Comment by jakubmisek** [comment]
Is there anything common with those 5 problems? Are those files just opened at the moment?

**Comment by menturion** [comment]
As far as I tested, all shown 'errors' are PHP2439 issues and the related files cannot be opened. 
All these errors have not been shown in v1.47.15538.

**Comment by jakubmisek** [comment]
@menturion thank you, but I'm asking completely different questions :)

**Comment by menturion** [comment]
Sorry, what I meant by my reply is that these 5 problems have in common that they are not of type PHP2439.

**Comment by menturion** [comment]
What do you mean by "Are those files just opened at the moment?".

Not at the moment, but I can open them. Should I?

**Comment by jakubmisek** [comment]
thank you, that's what I needed to know. I appreciate your help!

**Comment by menturion** [comment]
If this problem is caused by other extensions, why does it work when I downgrade phptools e.g. to v1.47.15512?

**Comment by jakubmisek** [comment]
> If this problem is caused by other extensions, why does it work when I downgrade phptools e.g. to v1.47.15512?

We have added/improved this "override" diagnostic around 1.47 version, but the issue here is the wrong path to the file. So I'm just trying to figure this out, any information may help.

**Comment by menturion** [comment]
Just tested. 
I am getting this issue also on macOS.

**Comment by menturion** [comment]
If it was a path issue on Windows OS, this issue shouldn't occur on macOS anymore. It is a fresh new installation on a MacBook M4 with only the min required set of extensions.

**Comment by jakubmisek** [comment]
@menturion - Please answer my question so I can help you.

**Comment by jakubmisek** [comment]
> If it was a path issue on Windows OS, this issue shouldn't occur on macOS anymore. It is a fresh new installation on a MacBook M4 with only the min required set of extensions.

does it mean the issue is fixed, and it was caused by file system issue on your PC?

**Comment by menturion** [comment]
> @menturion - Please answer my question so I can help you.
> 

I will as soon as I sit in front of the PC again.

**Comment by menturion** [comment]
> > If it was a path issue on Windows OS, this issue shouldn't occur on macOS anymore. It is a fresh new installation on a MacBook M4 with only the min required set of extensions.
> 
> does it mean the issue is fixed, and it was caused by file system issue on your PC?

It is not fixed, neither on Windows nor on macOS.
